src/PluginStartupRegistration.cpp

```diff
@@ -52,23 +53,32 @@ const std::vector<wxString>& PluginStartupRegistration::GetFailedPlugins() const
 
 void PluginStartupRegistration::Run()
 {
-   auto dialog = BasicUI::MakeProgress(XO("Searching for plugins"), XO(""));
-   ProcessNext();
-   while(mValidator)
+   try
+   {
+      auto dialog = BasicUI::MakeProgress(XO("Searching for plugins"), XO(""));
+      ProcessNext();
+      while(mValidator)
+      {
+         const auto message = TranslatableString { mPluginsToProcess[mCurrentPluginIndex].second, { } };
+         //Update UI
+         if(dialog->Poll(mCurrentPluginIndex, mPluginsToProcess.size(), message) != BasicUI::ProgressResult::Success)
+            Stop();
+         //AsyncPluginValidator uses event loop for internal message
+         //delivery, but ProgressDialog::Poll implementation does not call
+         //wxApp::Yield each time, which may result in too long CPU stalls
+         if(wxTheApp->HasPendingEvents())
+            wxTheApp->Yield();
+         else
+            //Seems like we do have no events yet, save CPU cycles then
+            std::this_thread::sleep_for(std::chrono::milliseconds { 10 });
+      }
+   }
+   catch(std::exception& e)
```

Comment:
Was this catch meant for the new throw?  But it does not enclose it.

Reply:
I don't see any other places from where StartHost could be called